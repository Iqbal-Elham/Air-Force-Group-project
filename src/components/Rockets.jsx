import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Rocket from './Rocket';
import { getRockets } from '../redux/rockets/rocketsSlice';

const Rockets = () => {
  const { allRockets } = useSelector((store) => store.rockets);
  const dispatch = useDispatch();
  useEffect(() => {
    if (!allRockets.length) dispatch(getRockets());
  }, [dispatch, allRockets]);

  return (
    <div>
      {allRockets.map((rocket) => (
        <Rocket
          key={rocket.id}
          id={rocket.rocket_id}
          name={rocket.rocket_name}
          type={rocket.rocket_type}
          img={rocket.flickr_images}
          description={rocket.description}
          reserved={rocket.reserved}
        />
      ))}
    </div>
  );
};

export default Rockets;
