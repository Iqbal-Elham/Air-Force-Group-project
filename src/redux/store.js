import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './rockets/rocketsSlice';
import missionsReducer from './missions/missionsSlice';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
    rockets: rocketReducer,
  },
});

export default store;
