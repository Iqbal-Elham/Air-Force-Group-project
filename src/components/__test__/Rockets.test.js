import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import Rockets from "../Rockets";
import { Provider } from "react-redux";
import store from "../../redux/store";

describe("Test the Rockets Component", () => {
    it("should match to snapshot", () => {
        const tree = renderer.create(
          <Provider store={store}>
              <Rockets />
          </Provider>
        );
        expect(tree).toMatchSnapshot();
      });
})