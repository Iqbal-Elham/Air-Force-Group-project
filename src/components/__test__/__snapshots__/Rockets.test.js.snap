// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Rockets Component should match to snapshot 1`] = `<div />`;
